app/src/main/java/emu/skyline/preference/IntegerListPreference.kt

```diff
@@ -0,0 +1,329 @@
+
+            /**
+             * Retrieve a singleton instance of this simple
+             * [androidx.preference.Preference.SummaryProvider] implementation.
+             *
+             * @return a singleton instance of this simple
+             * [androidx.preference.Preference.SummaryProvider] implementation
+             */
+            val instance : SimpleSummaryProvider?
+                get() {
+                    if (simpleSummaryProvider == null) {
+                        simpleSummaryProvider = SimpleSummaryProvider()
+                    }
+                    return simpleSummaryProvider
+                }
+        }
+    }
+
+    class IntegerListPreferenceDialogFragmentCompat : PreferenceDialogFragmentCompat() {
+        var clickedDialogEntryIndex = 0
+        private var entries : Array<CharSequence>? = null
+        private var entryValues : IntArray? = null
+
+        override fun onCreate(savedInstanceState : Bundle?) {
+            super.onCreate(savedInstanceState)
+            if (savedInstanceState == null) {
+                val preference = listPreference
+                check(preference.entries != null) { "IntegerListPreference requires at least the entries array." }
+                clickedDialogEntryIndex = preference.findIndexOfValue(preference.value)
+                entries = preference.entries
+                entryValues = preference.entryValues
+            } else {
+                clickedDialogEntryIndex = savedInstanceState.getInt(SAVE_STATE_INDEX, 0)
+                entries = savedInstanceState.getCharSequenceArray(SAVE_STATE_ENTRIES)
+                entryValues = savedInstanceState.getIntArray(SAVE_STATE_ENTRY_VALUES)
+            }
+        }
+
+        override fun onSaveInstanceState(outState : Bundle) {
+            super.onSaveInstanceState(outState)
+            outState.putInt(SAVE_STATE_INDEX, clickedDialogEntryIndex)
+            outState.putCharSequenceArray(SAVE_STATE_ENTRIES, entries)
+            outState.putIntArray(SAVE_STATE_ENTRY_VALUES, entryValues)
+        }
+
+        private val listPreference : IntegerListPreference
+            get() = preference as IntegerListPreference
+
+        override fun onPrepareDialogBuilder(builder : AlertDialog.Builder) {
+            super.onPrepareDialogBuilder(builder)
+            builder.setSingleChoiceItems(
+                entries, clickedDialogEntryIndex
+            ) { dialog, which ->
+                if (clickedDialogEntryIndex != which) {
+                    clickedDialogEntryIndex = which
+                    if (listPreference.refreshRequired)
+                        context?.getSettings()?.refreshRequired = true
+                }
+
+                // Clicking on an item simulates the positive button click, and dismisses
+                // the dialog.
+                onClick(
+                    dialog,
+                    DialogInterface.BUTTON_POSITIVE
+                )
```

Comment:
This is awful and excessive, not everything needs to be spaced out.. It is perfectly readable on a single line and if anything this makes it harder to read.